engines/mads/mps_installer.h

```diff
@@ -0,0 +1,94 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef MADS_MPS_INSTALLER_H
+#define MADS_MPS_INSTALLER_H
+
+#include "common/archive.h"
+#include "common/ptr.h"
+#include "common/stream.h"
+#include "common/hashmap.h"
+#include "common/hash-str.h"
+
+namespace MADS {
+
+#include "common/pack-start.h"  // START STRUCT PACKING
+
+struct FileDescriptorBin {
+	char name[0x52]; // zero-terminated, rest is filled with what looks like garbage
+	uint16 compression;
+	uint16 volumeNumber;
+	uint32 offsetInVolume;
+	uint32 compressedSize;
+	uint32 uncompressedSize;
+} PACKED_STRUCT;
```

Comment:
Why do we need this? It would be easier to just have FileDescriptor, and read into it with native endianess. This would simplify the code considerably, and we can remove FileDescriptorBin altogether